### ISSUE TITLE
Accessing and driving the agent arm joint-by-joint

### ISSUE THREAD

**Issue by vishal-2000** [allenai/manipulathor#10]
Hi,
We are working on a particular problem that requires us to manipulate/drive the arm joint-by-joint or gain access to the IK plans generated for every manipulation. We are working with Ai2thor's Manipulathor for this project, and we found the following snippet in the paper - https://arxiv.org/pdf/2104.11213.pdf (ManipulaTHOR: A Framework for Visual Object Manipulation):
" The robot’s arm rig has been designed to work with either forward or inverse kinematics (IK), meaning its motion can be driven joint-by-joint, or directly from the wrist, respectively. ".
Though later, the same paper pointed out that the API only provides wrist control to the user, I feel the aforementioned snippet hints at some workaround that we can use to achieve joint control.
So, basically:
1. How to access the ai2thor's default IK solver's solution for every API call like pick_object, release_object, and moveArm? (Expecting each such solution is a trajectory with coordinates for each of the arm's joints, similar to that of a classic IK solver)
and,
2. Is it possible to control these joints directly, like building a custom IK solver and sending the solution generated by this solver to the arm for execution? If so, in what way can we access these joints and drive the arm joint-by-joint?

We are currently using Ai2Thor version 3.3.4.

**Comment by ehsanik** [comment]
Hi,

1. You can get the solution for every call in the metadata. It is saved in the event that is returned after calling a function. More specifically, you can access it using the following command:
```
event = controller.step('MoveArmBase', y=0.2)
joints = event.metadata['arm']['joints'] # len(joints) = 4, where the first element shows transformations and rotations for the joint closest to the agent's body and the last one represents the gripper's info.
```
2. We currently do not support custom IK solvers due to the limitations of Unity API. However, we are working on adding this feature in the future. 

Let me know if you have any other questions.
